test/spectral/tests/custom-rules/request-body-object.test.js

```diff
@@ -0,0 +1,55 @@
+const inCodeValidator = require('../../../../src/lib');
+
+describe('spectral - test that request body schema is an object', function() {
+  it('should error only when request body is not an object', async () => {
+    const spec = {
+      openapi: '3.0.0',
+      paths: {
+        '/path1': {
+          post: {
+            operationId: 'addPet',
+            requestBody: {
+              content: {
+                'application/json': {
+                  schema: {
+                    // no error
+                    type: 'object',
+                    properties: {
+                      prop1: {
+                        type: 'string'
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          },
+          put: {
+            operationId: 'updatePet',
+            requestBody: {
+              content: {
+                'application/json': {
+                  schema: {
+                    // error 1
+                    type: 'array',
+                    items: {
+                      type: 'string'
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    };
```

Comment:
Please add one more test -- where the request body schema has `properties` but _not_ `type: object`.

I think (I _hope_) that this will not trigger a message, but we should make sure.

Reply:
The core Spectral `enumeration` function does not catch the case where the `type` field is omitted.

I will create a custom function that catches this.

Reply:
Fixed

Reply:
I think you misunderstood my comment.  I don't think we should require `type: object` -- it is generally assumed if there are properties. And perhaps we could check for `properties` if there is no `type` field, but I'd rather just keep this simple.  If `type` is present, make sure it is `object`, and and if not issue a message.  if `type` is not present, then don't issue any message.

Reply:
I did misunderstand the comment. I returned the rule to its original state and added a test to ensure that no error is issued when `type` is omitted.